lib/puma/request.rb

```diff
@@ -104,6 +108,12 @@ def handle_request(client, requests)
           status, headers, app_body = [501, {}, ["#{env[REQUEST_METHOD]} method is not supported"]]
         end
 
+        if protocol = headers.delete(RACK_PROTOCOL)
+          status = 101
```

Comment:
I think it's advisable we force the status code here. HTTP/2 uses `200 Ok` but HTTP/1 uses 101... There is no reason to use anything other than 101 AFAIK.